api/nakadi-event-bus-api.yaml

```diff
@@ -881,6 +891,47 @@ paths:
           format: int32
           required: false
           default: 10000
+        - name: commit_mode
+          in: query
+          description: |
+            The commit mode that will be used for this stream. Currently only `commit_at_will` is supported.
+            Possible values:
+
+            - `commit_at_will`: client is free to choose how often he wants to commit. Of course the commit_timeout
+            property should still be respected.
+            When a batch is committed that also automatically commits all previous batches that were sent in a stream
+            (if the commit_semantics parameter is `per_partition` - that will commit all previous batches only for
+            current partition).
+
+            - `commit_each`: this commit mode requires each batch to be committed, and it should be done in correct
+            order. Committing a cursor when the previous is not yet committed is not possible.
+
+            - `auto_commit`: everything that Nakadi sends to stream is considered to be committed. No actions are
+            required from client side to commit. Be aware that with this commit mode loosing of some events is very
+            probable. So you should know what you are doing if you use autocommit.
+          type: string
+          enum:
+            - commit_at_will
+            - commit_each
+            - auto_commit
+          required: false
+          default: commit_at_will
+        - name: commit_semantics
+          in: query
+          description: |
+            The semantics of commit that will be used for this stream. Possible values:
+
+            - `per_partition`: Each partition is committed separatelly. Each cursor in a stream belongs to one of the
+            partitions of that stream; committing that cursor will commit the data only for that partition.
+
+            - `per_stream`: When you commit a cursor - all data in a stream before this cursor is
```

Comment:
This semantics is really hard to understand. There is no relation of order between partitions so it really doesn't make sense to say that it will commit something from other partition that is "previous" to this cursor.